core/amber/src/main/scala/edu/uci/ics/amber/engine/architecture/worker/ChannelMarkerManager.scala

```diff
@@ -31,14 +32,20 @@ import edu.uci.ics.amber.core.virtualidentity.{
   ChannelIdentity,
   ChannelMarkerIdentity
 }
+import edu.uci.ics.amber.core.workflow.PortIdentity
 
 import scala.collection.mutable
 
-class ChannelMarkerManager(val actorId: ActorVirtualIdentity, inputGateway: InputGateway)
-    extends AmberLogging {
+class ChannelMarkerManager(
+    val actorId: ActorVirtualIdentity,
+    inputGateway: InputGateway,
+    inputManager: InputManager
+) extends AmberLogging {
 
   private val markerReceived =
-    new mutable.HashMap[ChannelMarkerIdentity, Set[ChannelIdentity]]()
+    new mutable.HashMap[ChannelMarkerIdentity, mutable.HashMap[PortIdentity, Set[
```

Comment:
Do you need to change this variable? I think you only need one helper function to get the port of a given channel.

Reply:
Removed the change of markerReceived.